src/libbson/src/bson/bson-macros.h

```diff
@@ -293,6 +293,9 @@
 #define BSON_GNUC_DEPRECATED
 #endif
 
+#define BSON_CONCAT_1(a, ...) a##__VA_ARGS__
+#define BSON_CONCAT(a, ...) BSON_CONCAT_1 (a, __VA_ARGS__)
+#define BSON_CONCAT3(a, b, c) BSON_CONCAT (a, BSON_CONCAT (b, c))
```

Comment:
I do not recall off-hand the rules around token concatenation and macro evaluation, so I may be misunderstanding the need for `BSON_CONCAT_1`, but should this have an underscore for consistency?

```suggestion
#define BSON_CONCAT_3(a, b, c) BSON_CONCAT (a, BSON_CONCAT (b, c))
```

Reply:
The `BSON_CONCAT` deferring to `BSON_CONCAT_1` is a safety-net for a whackiness in the way macro expansion works ([Refer](https://gcc.gnu.org/onlinedocs/cpp/Argument-Prescan.html#Argument-Prescan)). The `_1` is supposed to say "Don't call me directly," so `_IMPL` might be a better suffix. The `3` is just "I concatenate three tokens."

Reply:
Thanks for the link. `_IMPL` does seem like a clearer suffix.